### DISCUSSION TITLE
Gif support for the <Image> tag

### DISCUSSION THREAD

**Discussion by unisteven** [vercel/next.js#23181] [Ideas]
### Describe the feature you'd like to request

At this moment Nextjs has support for automatically optimising png or jpeg images with the option to set the quility. During testing I found out that there was no support for .gif images, it will just serve the original file.
I would like to see Gifs supported with the image optimalisation tool. That includes animated gifs.

### Describe the solution you'd like

A more modern standard is WebP which also supports animated images. It would be nice to be able to set the quility for each image and that it will automatically create a WebP optimized image. 

### Describe alternatives you've considered

Currently I use a external system on my api that transforms the images on the serverside. This is done through a third party image api.

**Comment by timneutkens** [comment]
Gifs are very tricky to optimize and in most cases we found they are reduced in quality tremendously when optimized or the gif animation order would not be correct anymore. Hence why it's not supported. We even saw many cases where gifs would increase in size when optimized.

**Reply by unisteven** [reply]
Allright, thanks for the information. Maybe it would be nice to update the documentation (https://nextjs.org/docs/basic-features/image-optimization) to have a list of supported formats and a small explainer why gifs are not supported.

**Reply by timneutkens** [reply]
Feel free to send a PR to update the docs 👍

**Comment by lhguerra** [comment]
Can't next/image even resize gifs by default?

**Comment by antho1404** [comment]
One good way to optimize GIFs is to stream them as a video. This would still more or less preserve the quality of the gif, reduce the size, and allow streaming that will be pretty much immediate rendering instead of waiting to download a 10Mb file, for example.

This would be amazing to have in the `next/image` even tho this is not rendering an image anymore but a video.

**Comment by adamJLev** [comment]
Sharp makes it very easy to transform animated gif's to animated webp, a widely supported format now. 
This feature should probably be reconsidered for implementation now
